operators/src/processing/raster_vector_join/points.rs

```diff
@@ -1,139 +1,252 @@
+            if value_option.is_some() {
+                continue; // already has a value
             }
+
+            let value = match raster.pixel_value_at_coord(coordinate) {
+                Ok(value) => value,
+                Err(_) => continue, // not found in this raster tile
+            };
+
+            if raster.is_no_data(value) {
+                continue; // TODO: is it okay to treat no data the same as not found?
+            }
+
+            *value_option = Some(value);
         }
 
-        points
-            .add_column(new_column_name, aggregator.into_data())
-            .map_err(Into::into)
+        Ok(self)
     }
 
-    fn create_aggregator<P: Pixel>(
-        number_of_features: usize,
-        aggregation: AggregationMethod,
-    ) -> TypedAggregator {
-        match aggregation {
-            AggregationMethod::First => match P::TYPE {
-                RasterDataType::U8
-                | RasterDataType::U16
-                | RasterDataType::U32
-                | RasterDataType::U64
-                | RasterDataType::I8
-                | RasterDataType::I16
-                | RasterDataType::I32
-                | RasterDataType::I64 => {
-                    FirstValueDecimalAggregator::new(number_of_features).into_typed()
-                }
-                RasterDataType::F32 | RasterDataType::F64 => {
-                    FirstValueNumberAggregator::new(number_of_features).into_typed()
+    fn into_colletion(self, new_column_name: &str) -> Result<MultiPointCollection> {
+        let points = match &self.points {
+            Some(points) => points,
+            None => return Err(Error::EmptyInput), // TODO: maybe output empty dataset or just nulls
+        };
+
+        // TODO: directly save values in vector of correct type
+        // TODO: handle classified values
+        let feature_data = match P::TYPE {
+            RasterDataType::U8
+            | RasterDataType::U16
+            | RasterDataType::U32
+            | RasterDataType::U64
+            | RasterDataType::I8
+            | RasterDataType::I16
+            | RasterDataType::I32
+            | RasterDataType::I64 => FeatureData::NullableDecimal(self.typed_values(points)),
+            RasterDataType::F32 | RasterDataType::F64 => {
+                FeatureData::NullableNumber(self.typed_values(points))
+            }
+        };
+
+        Ok(points.add_column(new_column_name, feature_data)?)
+    }
+
+    fn typed_values<To: Pixel>(&self, points: &MultiPointCollection) -> Vec<Option<To>>
+    where
+        P: AsPrimitive<To>,
+    {
+        points
+            .feature_offsets()
+            .windows(2)
+            .map(|window| {
+                let (begin, end) = match *window {
+                    [a, b] => (a as usize, b as usize),
+                    _ => return None,
+                };
+
+                let slice = &self.values[begin..end];
+
+                // TODO: implement aggregation methods
```

Comment:
its a new task i guess?

Reply:
Yes, I already added it:

https://trello.com/c/iqYOGYuJ/92-innere-aggregation-bei-raster-vektor-join